c/tskit/genotypes.c

```diff
@@ -474,20 +470,31 @@ tsk_variant_get_allele_index(tsk_variant_t *self, const char *allele, tsk_size_t
     return ret;
 }
 
-static int
-tsk_variant_update_site(tsk_variant_t *self)
```

Comment:
Might be better to keep this as tsk_variant_update_site() and then just keep ``tsk_tree_get_variant`` as simple as possible (just a call to this method)?

Much of the functionality here really does feel like a "method" of the variant object, so it feels artificial/wrong putting it into the tree class.